### PR TITLE
PRODUCTION

### PR DESCRIPTION
**Pull request by divinecharlotte**
## IN THIS PR I IMPLEMENTED THE FOLLOWING:

- Splash screen

-A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and login pages

1. - The user should be able to register in the app with full name, email and password (all mandatory).
2. - The user can log into the app using email and password.
3. - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)

1. When the user logs in, they are presented with the categories page.
2. For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
3. When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
4. There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

-  Transactions page

1. - For a given category, the list of transactions is presented, ordered by the most recent.
2. At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
3. There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
4. When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

1. The user fills out a form to create a new category, indicating their name and icon (both mandatory).
2. The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
3. When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
The user fills out a form to create a new transaction with:
1. name (mandatory)
2. amount (mandatory)
3. categories (mandatory at least one)
5. The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
6. When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,61 @@
-# Budget-App
\ No newline at end of file
+## BUDGET APP
+>Is a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what
+
+## lIVE DEMO
+- This is the link to the website:[BUDGET APP]()
```
Comment (Whoistolu) [inline README.md]: - [ ] [REQUIRED] Kindly ensure that you deploy the application using Heroku, Netlify, or GitHub Pages. Please make sure that the home and detail page works as expected. Currently, it looks like you mistakenly attached the URL wrong as the live version `https://github.com/divinecharlotte/Budget-App/blob/dev`. Please fix this as the live version will help people have quick access to the app.

File: README.md [hunk 2]
```diff
@@ -1 +1,61 @@
-# Budget-App
\ No newline at end of file
+## BUDGET APP
+>Is a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what
```
Comment (Whoistolu) [inline README.md:2]: - [ ] [REQUIRED] I see you have attached the video presentation on the PR description as project documentation. Let's also add the link to the README as it is the file giving the entire documentation of the project. 
Also, one of the project requirements states that you should `highlight a piece of the code` that you are proud of making. Kindly make another video to correct this.

File: app/icons/icons.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[
+    {"name": "Birthday","url": "https://i.pinimg.com/originals/58/8f/3a/588f3a61e05370ee6a0175c7012c4ce0.jpg"},
+    {"name": "Shopping","url": "https://insideretail.com.au/wp-content/uploads/2020/11/InstaNightShopping.png"},
+    {"name": "Education","url": "https://png.pngitem.com/pimgs/s/292-2923379_undergraduate-education-logo-for-education-png-transparent-png.png"},
+    {"name": "Vacation","url": "https://image.similarpng.com/very-thumbnail/2021/08/Travel-agency-logo-design-template-on-transparent-background-PNG.png"},
+    {"name": "Furniture","url": "https://www.creativefabrica.com/wp-content/uploads/2021/02/16/Elegant-furniture-logo-with-golden-couch-Graphics-8659511-1-1-580x386.png"}
+  ]
```
Comment (Whoistolu) [inline app/icons/icons.json:7]: - [ ] Excellent implementation on the category dropdowns 👍🏽

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @divinecharlotte 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — CHANGES_REQUESTED** [review]
Hello @divinecharlotte , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Project is deployed well and working  👏
- Professional Readme file  👏

 ## Required Changes ♻️

_ [Required] As mentioned by the previous reviewer , In your description video you should highlight(talk about or explain) a piece of code that you are proud of making. Kindly fix this issue 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by cwaku — APPROVED** [review]
Hi @divinecharlotte 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - [x] Good job overall following the template design 💯 
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**